src/extension.ts

```diff
@@ -424,3 +423,33 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         context.subscriptions.push(createClient(dirToAnalyze).start());
     }
 }
+
+// Search all the workspace folders and return the first that contains .phan/config.php
+// This will return the folder name in an array as required, and just only one folder,
+// the first met. We can easy modify the function to add all valid workspace folders to the array.
+function findValidProjectDirectories(): string[] {
+    // Get the fsPath(file system path) of all workspace folders.
+    const VSCodeFolders = vscode.workspace.workspaceFolders;
+    if (!VSCodeFolders) {
+        return [];
+    }
+    let workingFolders = VSCodeFolders.map(function (obj) {
+        if (('uri' in obj) && ('fsPath' in obj.uri)) {
+            return obj.uri.fsPath;
+        }
+        return '';
```

Comment:
This would be clearer as a foreach loop - For example, this should filter out the objects without fsPath instead of using the empty string (Is that treated like the current working directory? I can't tell)

Reply:
Yeah,I think that the if is unnecessary, but I did that to be sure to not throw an exception accessing a undefined property. Changed to:
```
        const filePathExistInObject = ( ('uri' in obj) && ('fsPath' in obj.uri) );
        if ( filePathExistInObject ) {
            return obj.uri.fsPath;
        }
        return '';
```

Reply:
I mean this would be clearer as a for loop (written in an imperative style) instead of a call to map() and filter()

Reply:
I changed to just "return obj.uri.fsPath;", since "uri" will always be in "WorkspaceFolder", check [here](https://code.visualstudio.com/api/references/vscode-api#WorkspaceFolder), and "fsPath" will always be in uri: [here](https://code.visualstudio.com/api/references/vscode-api#Uri)